### PR TITLE
Fix all 5 swim issue and add support for floorNumber in site_intent m…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…odule and write docs for each api as well.

SWIM issues were - 

- Cannot tag/distribute/activate image individually without importing image in the same playbook
- Failed when importing an existing image
- NoneType AttributeError error when importing a new image 
- Parameter deviceIPAddress (document) not match parameter deviceIpAddress (module)
- The siteName within the 'specified site or list of sites' is not functioning correctly

Site Intent - 

-- Give the support for floorNumber while creating updating Floor as part of site intent module.

### CODE REVIEW

File: plugins/modules/swim_intent.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from __future__ import absolute_import, division, print_function
 
 __metaclass__ = type
-__author__ = ("Madhan Sankaranarayanan, Rishita Chowdhary, Abhishek Maheshwari")
+__author__ = ("Madhan Sankaranarayanan, Abhishek Maheshwari")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:10]: Please kindly don't remove existing authors..

File: plugins/modules/swim_intent.py [hunk 2]
```diff
@@ -27,7 +27,6 @@
 extends_documentation_fragment:
   - cisco.dnac.intent_params
 author: Madhan Sankaranarayanan (@madhansansel)
-        Rishita Chowdhary (@rishitachowdhary)
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:30]: Please kindly don't remove existing authors..

File: plugins/modules/swim_intent.py [hunk 3]
```diff
@@ -615,10 +618,25 @@ def get_diff_import(self):
...
             image_name = self.want.get("url_import_details").get("payload")[0].get("sourceURL")
+
+            # Code to check if image already exist in the DNAC
+            name = image_name.split('/')[-1]
+            image_exist = self.is_image_exist(name)
+            if image_exist:
+                image_id = self.get_image_id(image_name)
+                self.have["imported_image_id"] = image_id
+                self.result['msg'] = "Image already exists."
+                log("Image already exists in the Cisco DNA Center")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:636]: Can we print with Image name?

File: plugins/modules/swim_intent.py [hunk 4]
```diff
@@ -615,10 +618,25 @@ def get_diff_import(self):
...
         if self.want.get("import_type") == "url":
             image_name = self.want.get("url_import_details").get("payload")[0].get("sourceURL")
+
+            # Code to check if image already exist in the DNAC
+            name = image_name.split('/')[-1]
+            image_exist = self.is_image_exist(name)
+            if image_exist:
+                image_id = self.get_image_id(image_name)
+                self.have["imported_image_id"] = image_id
+                self.result['msg'] = "Image already exists."
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:635]: Add image name as well..

File: plugins/modules/swim_intent.py [hunk 5]
```diff
@@ -633,6 +651,17 @@ def get_diff_import(self):
             )
         else:
             image_name = self.want.get("local_import_details").get("filePath")
+            # Code to check if image already exist in the DNAC
+            name = image_name.split('/')[-1]
+            image_exist = self.is_image_exist(name)
+            if image_exist:
+                image_id = self.get_image_id(name)
+                self.have["imported_image_id"] = image_id
+                self.result['msg'] = "Image already exists."
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:660]: Add Image name

File: plugins/modules/swim_intent.py [hunk 6]
```diff
@@ -633,6 +651,17 @@ def get_diff_import(self):
...
+                log("Image already exists in the Cisco DNA Center")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:661]: Add Image name

File: plugins/modules/swim_intent.py [hunk 7]
```diff
@@ -657,21 +686,22 @@ def get_diff_import(self):
             if task_details and \
                     ("completed successfully" in task_details.get("progress").lower()):
                 self.result['changed'] = True
-                self.result['msg'] = "Image imported successfully"
+                self.result['msg'] = "Swim Image imported successfully"
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:689]: Add Image name

File: plugins/modules/swim_intent.py [hunk 8]
```diff
@@ -657,21 +686,22 @@ def get_diff_import(self):
...
+                log("Swim Image imported successfully")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:690]: Add Image name

File: plugins/modules/swim_intent.py [hunk 9]
```diff
@@ -657,21 +686,22 @@ def get_diff_import(self):
...
                 break
 
             if task_details and task_details.get("isError"):
-                if "Image already exists" in task_details.get("failureReason"):
+                if "already exists" in task_details.get("failureReason"):
                     self.result['msg'] = "Image already exists."
+                    log("Image already exists in the Cisco DNA Center")
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:696]: Add Image name

File: plugins/modules/swim_intent.py [hunk 10]
```diff
@@ -657,21 +686,22 @@ def get_diff_import(self):
...
                 self.result['changed'] = True
-                self.result['msg'] = "Image imported successfully"
+                self.result['msg'] = "Swim Image imported successfully"
+                log("Swim Image imported successfully")
                 break
 
             if task_details and task_details.get("isError"):
-                if "Image already exists" in task_details.get("failureReason"):
+                if "already exists" in task_details.get("failureReason"):
                     self.result['msg'] = "Image already exists."
```
Comment (madhansansel) [inline plugins/modules/swim_intent.py:695]: Add Image name

File: plugins/modules/swim_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:10]: sure, it's remove by mistake.

File: plugins/modules/swim_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:636]: addressed

File: plugins/modules/swim_intent.py (hunk 5 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:660]: added

File: plugins/modules/swim_intent.py (hunk 7 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:689]: Added

File: plugins/modules/swim_intent.py (hunk 8 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:690]: added

File: plugins/modules/swim_intent.py (hunk 9 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:696]: added

File: plugins/modules/swim_intent.py (hunk 10 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:695]: added

File: plugins/modules/swim_intent.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:635]: added

File: plugins/modules/swim_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/swim_intent.py:30]: reverted the change

**Review by madhansansel — APPROVED** [review]

**Pull request merged**